.github/workflows/rpm-build.yaml

```diff
@@ -4,17 +4,40 @@
 ---
 name: Build DecisionEngine Modules RPM
 
-on: [pull_request, push]
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: "branch ref"
+        required: true
+        default: "master"
 
 jobs:
   rpmbuild_el7:
     name: Build an EL7 rpm
     runs-on: ubuntu-latest
     steps:
+      - name: make date tag
+        id: mkdatetag
+        run: echo "::set-output name=dtag::$(date +%Y%m%d_%H%M%S)"
+
+      - name: make ref tag case 1
+        id: mkreftag1
+        if: ${{ github.event.inputs.ref != '' }}
+        run: echo "::set-output name=reftag::${{github.event.inputs.ref}}"
+
+      - name: make ref tag case 2
+        id: mkreftag2
+        if: ${{ github.event.inputs.ref == '' }}
+        run: echo "::set-output name=reftag::${GITHUB_BASE_REF:+PR}"$(awk -F"/" '{print $3}' <<< ${GITHUB_REF})
+
       - name: Checkout code tree
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
+          ref: ${{github.event.inputs.ref}}
```

Comment:
Will this work also when `github.event.inputs.ref` is `""`?  Do you have to protect w/ an if?
Can the ref be ever an empty string?
- with workflow_dispatch there is a default
- with PR and push there should be the branch?

Reply:
From the [checkout action usage](https://github.com/actions/checkout/tree/releases/v2#usage)
```
# The branch, tag or SHA to checkout. When checking out the repository that
# triggered a workflow, this defaults to the reference or SHA for that event.
# Otherwise, uses the default branch.
ref: ''
```
in case `github.event.inputs.ref` is not set, because the action is triggered by a push to the PR or to a branch, `ref` is set to the references of the event that triggered the action.
I tested both cases in my fork and I confirmed this behavior.
In the case the action is triggered by `workflow_dispatch` event, as you noted, the behavior is more explicit.